src/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -86,6 +86,22 @@ private static unsafe void InitializeGlobalTablesForModule(IntPtr moduleManager)
                 Debug.Assert(length % IntPtr.Size == 0);
                 InitializeStatics(staticsSection, length);
             }
+
+            // Initialize frozen object segment with GC present
+            IntPtr frozenObjectSection = GetModuleSection(moduleManager, ReadyToRunSectionType.FrozenObjectRegion, out length);
+            if (frozenObjectSection != IntPtr.Zero)
+            {
+                InitializeFrozenObjectSegment(frozenObjectSection, length);
```

Comment:
It means that FrozenObjectSentinelNode is not getting emitted correctly. It should guarantee the alignment.